transport/http/outbound.go

```diff
@@ -41,61 +45,61 @@ var (
 	errOutboundNotStarted     = errors.ErrOutboundNotStarted("http.Outbound")
 )
 
-type outboundConfig struct {
-	keepAlive time.Duration
-}
+// NewOutbound is deprecated
+// TODO rename this func NewOutboundFromURL
+func NewOutbound(urlStr string) transport.Outbound {
```

Comment:
Also, this is actually making a breaking change to `NewOutbound` by removing
its Options varargs. Consider adding them back somehow. You can share options
between things by using interfaces/implementing multiple functions:

```
type Option interface {
    applyOutboundOption(*outboundConfig)
    applyAgentOption(*agentConfig)
}

func KeepAlive(time.Duration) Option {
    ...
}

func NewAgent(opts ...Option) *Agent {
    // ..
}

func NewOutbound(url string, opts ...Option) Outbound {
    // ..
}
```